### ISSUE TITLE
[Bug]: HTTP V2 restful Return "code":0

### ISSUE THREAD

**Issue by syang1997** [milvus-io/milvus#35969]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.4.6-v2.4.10
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):     pulsar
- SDK version(e.g. pymilvus v2.0.0rc2): http
- OS(Ubuntu or CentOS): CentOS
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

List and search and other interfaces response code : 0

curl --location --request POST "http://${MILVUS_URI}/v2/vectordb/collections/list" \
> --header "Content-Type: application/json" \
> --header "Authorization: Bearer ${TOKEN}" \
> --data-raw '{
>     "dbName": "default"
> }'
{"code":0,"data":["trend_nlp3"]}

**Comment by syang1997** [comment]
https://milvus.io/api-reference/restful/v2.4.x/v2/Collection%20(v2)/List.md

<img width="863" alt="image" src="https://github.com/user-attachments/assets/685f934e-4223-45c6-affc-3e9df760b87b">

**Comment by xiaofan-luan** [comment]
/assign @smellthemoon 
can you help on this?

**Comment by xiaofan-luan** [comment]
I think all the succesfully request should return 200

**Comment by zhuwenxing** [comment]
> I think all the succesfully request should return 200

This feature is intentional by design, and it was supposed to be aligned with the kernel's code at the time.
The document should be updated

cc @czs007

**Comment by zhuwenxing** [comment]
see https://github.com/milvus-io/milvus/pull/33339
![image](https://github.com/user-attachments/assets/51860141-df01-414b-bfea-505ab6ec57aa)

**Comment by xiaofan-luan** [comment]
this will cause serious compatible issue?

**Comment by xiaofan-luan** [comment]
at least 0 and 200 should both mean successfuly

**Comment by zhuwenxing** [comment]
> this will cause serious compatible issue?

If the user uses the value of the "code" field to determine whether the request was successful, it will lead to incorrect judgments. In terms of functional usage, there will be no compatibility issues.

**Comment by xiaofan-luan** [comment]
this is the compatible issue, this is what people do and change the error code may nreak

resp := request()
if resp.code !=200 {
      return error.Warp(resp)
}

**Comment by jaime0815** [comment]
Use HTTP codes rather than Milvus error codes in the result API,  200 status code should be returned in this case.

**Comment by smellthemoon** [comment]
> Use HTTP codes rather than Milvus error codes in the result API, 200 status code should be returned in this case.

I agree with it. Change the milvus error codes back to HTTP codes?  Or it's necessary to add a extra milvus error code in the return list? @xiaofan-luan @czs007

**Comment by smellthemoon** [comment]
but if return code: 200, milvus err code also has 200 which refers to a specific error. Users can't know whether the req success based on the 200 code.

**Comment by xiaofan-luan** [comment]
can we seperate http response code and milvus error code?
milvus error code could be a field

**Comment by smellthemoon** [comment]
> can we seperate http response code and milvus error code?
> milvus error code could be a field

yes, we could do this. We will return a json, we can add both two err code if needed.

**Comment by smellthemoon** [comment]
HTTP code 200 indicates that the HTTP request is successful, which can be seen in the HTTP header field. The code in the returned JSON field is the code of Milvus. We will update the document. Thank you for your feedback! @syang1997

**Comment by yanliang567** [comment]
> > can we seperate http response code and milvus error code?
> > milvus error code could be a field
> 
> yes, we could do this. We will return a json, we can add both two err code if needed.

@smellthemoon have we fix this?

**Comment by smellthemoon** [comment]
> > > can we seperate http response code and milvus error code?
> > > milvus error code could be a field
> > 
> > 
> > yes, we could do this. We will return a json, we can add both two err code if needed.
> 
> @smellthemoon have we fix this?

After discussion, the final decision was to keep the existing behavior unchanged, that is, the Http code is in the http header and the milvus code is in the json. The document has been changed and emphasized.
